File: Lab-2/q1/app/src/main/java/com/example/q1/MainActivity.java
package com.example.q1;

import android.os.Bundle;
import android.util.Log;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.d(TAG, "onCreate() method called");
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.d(TAG, "onStart() method called");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.d(TAG, "onResume() method called");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.d(TAG, "onPause() method called");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.d(TAG, "onStop() method called");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.d(TAG, "onRestart() method called");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d(TAG, "onDestroy() method called");
    }
}

File: Lab-4/q5/app/src/main/java/com/example/food_items/MainActivity.kt
package com.example.food_items

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.CheckBox
import android.widget.ListView
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    private lateinit var  listView: ListView
    private lateinit var submit: Button
    private var orderSubmitted=false
    private val foodItems=arrayOf("Manchurian","Noodles","Fried Rice","Cake")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        listView=findViewById<ListView>(R.id.listview)
        submit=findViewById<Button>(R.id.submit)
        val adapter= ArrayAdapter<String>(
            this,
            R.layout.list_item,
            R.id.food_text,
            foodItems
        )
        listView.adapter=adapter
        submit.setOnClickListener {
            if(!orderSubmitted){
                displayOrder()
                disableListViewItems()
                orderSubmitted=true
            }
        }
    }
    private fun displayOrder(){
        val selectedItems= mutableListOf<String>()
        val adapter=listView.adapter as ArrayAdapter<*>
        for (i in 0 until adapter.count) {
            val item = adapter.getItem(i) as String
            val view = listView.getChildAt(i)
            val checkBox = view.findViewById<CheckBox>(R.id.checkbox_food)
            if (checkBox.isChecked) {
                selectedItems.add(item)
            }
        }
        val orderDetails = StringBuilder()
        val totalCost = selectedItems.sumOf { getFoodCost(it) }
        orderDetails.append("\nTotal Cost: $$totalCost")
        Toast.makeText(this, orderDetails.toString(), Toast.LENGTH_LONG).show()
    }
    private fun getFoodCost(foodItem: String): Int {
        // You can customize the cost based on your menu
        return when (foodItem) {
            "Manchurian" -> 30
            "Noodles" -> 45
            "Fried Rice" -> 20
            "Cake"->25
            else -> 0
        }
    }
    private fun disableListViewItems() {
        for (i in 0 until listView.childCount) {
            val view = listView.getChildAt(i)
            val checkBox = view.findViewById<CheckBox>(R.id.checkbox_food)
            checkBox.isEnabled = false
        }
    }
}